src/sentry/tasks/files.py

```diff
@@ -8,6 +8,7 @@
     queue="files.delete",
     default_retry_delay=60 * 5,
     max_retries=MAX_RETRIES,
+    autoretry_for=(DatabaseError, IntegrityError),
 )
```

Comment:
It might also be worth setting `acks_late=True` here. That way if the worker straight up crashes the task will be replayed.